container/src/columnation.rs

```diff
@@ -238,6 +240,27 @@ impl<T: Columnation> Clone for TimelyStack<T> {
     }
 }
 
+impl<T: Columnation> PushInto<TimelyStack<T>> for T {
+    #[inline]
+    fn push_into(self, target: &mut TimelyStack<T>) {
+        target.copy(&self);
+    }
+}
+
+impl<T: Columnation> PushInto<TimelyStack<T>> for &T {
+    #[inline]
+    fn push_into(self, target: &mut TimelyStack<T>) {
+        target.copy(self);
+    }
+}
+
+impl<T: Columnation> PushInto<TimelyStack<T>> for &&T {
+    #[inline]
+    fn push_into(self, target: &mut TimelyStack<T>) {
+        target.copy(self);
+    }
+}
```

Comment:
Maybe silly, but can this be `impl<T: PushInto<TimelyStack<T>> .. for &T`? Would that get both of these, and also `&&&T`?

Reply:
Yeah, this is a curiosity! We currently don't need in timely, but we might need it at some point. It arises when reading out of a container that returns references and copying into a new container. So while the items itself might be a reference, the reading can add another reference, which requires adding the `&&` implementation. Concretely, consider a container for slices, and we'd like to copy `[T]` in it. This means that the receiver needs to absorb `&T`, because that's the only thing we can get out of a slice. Now, we want `T` to be a container itself, and suddently we have nested references. This line in flatcontainer requires it:  (( You wouldn't be wrong to say it's quite esoteric, but, hey, it works :) ))
https://github.com/antiguru/flatcontainer/blob/main/src/lib.rs#L670